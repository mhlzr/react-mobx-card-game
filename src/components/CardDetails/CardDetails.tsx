import React, { ReactElement, FunctionComponent } from 'react';
import { observer, inject } from 'mobx-react';
import styled from 'styled-components';

import { Headline } from '../Headline/Headline';
import { CardGameStore } from 'src/stores/CardGameStore';

const StyledSection = styled.section`
    border: 1px solid var(--secondary-color);
    grid-area: details;
    padding: 1em;
`;

const StyledList = styled.ul`
    font-size: 1.5em;
    list-style: none;
    margin: 1em 0;
    padding: 0;
`;

const StyledListItem = styled.li`
    &:before {
        content: 'â€”';
        padding-right: 0.5em
    }
`

interface CardDetailsProps {
    store: CardGameStore;
}

export const CardDetails: FunctionComponent<CardDetailsProps> = inject('store')(observer(({ store }: CardDetailsProps): ReactElement | null => {
    const { player } = store;

    if (!player) {
        return null;
    }
    return (
        <StyledSection>
            <Headline>Details</Headline>
            <StyledList>
                <StyledListItem>{player.name}</StyledListItem>
                <StyledListItem>{player.alias}</StyledListItem>
                <StyledListItem>{player.asset}</StyledListItem>
            </StyledList>
        </StyledSection>
    )
}));
