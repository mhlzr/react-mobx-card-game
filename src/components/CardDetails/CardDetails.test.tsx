import React from 'react';
import { mount } from 'enzyme'

import { CardDetailsComponent } from './CardDetails';
import { cardGameStore } from '../../stores/__mock__/cardGameStore.mock';

describe('CardDetails', () => {

    it('does not render any content if there is no player selected', () => {
        cardGameStore.player = null;
        const wrapper = mount(<CardDetailsComponent store={cardGameStore} />);
        expect(wrapper.children()).toHaveLength(0)
    });

    it('renders if there is a player selected', () => {
        cardGameStore.player = cardGameStore.players[0];
        const wrapper = mount(<CardDetailsComponent store={cardGameStore} />);
        expect(wrapper.children()).toHaveLength(1);
    });

    it('has a snapshot', () => {
        cardGameStore.player = cardGameStore.players[0];
        const wrapper = mount(<CardDetailsComponent store={cardGameStore} />);
        expect(wrapper).toMatchSnapshot();
    });
})
