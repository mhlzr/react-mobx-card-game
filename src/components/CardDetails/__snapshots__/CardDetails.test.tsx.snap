// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails has a snapshot 1`] = `
<Component
  store={
    Object {
      "_players": Array [],
      "fetchPlayers": [MockFunction],
      "isFetching": false,
      "isSaving": false,
      "player": Player {
        "alias": "Dreamlurk The Unstoppable",
        "asset": "Foghammer Lead",
        "name": "Brianna Forbes",
      },
      "playerSortation": null,
      "players": Array [
        Player {
          "alias": "Dreamlurk The Unstoppable",
          "asset": "Foghammer Lead",
          "name": "Brianna Forbes",
        },
        Player {
          "alias": "Crystaldash",
          "asset": "Secret Glowquake Gold",
          "name": "Darcy Candice Ball",
        },
        Player {
          "alias": "Speedsoul",
          "asset": "Shifting Rainshadow Iron",
          "name": "Hillary Gibbs",
        },
        Player {
          "alias": "Seekvenom The Mystic",
          "asset": "Valkyries' Opal Adamant",
          "name": "Elva Becky Hammond",
        },
        Player {
          "alias": "Coincurse The Ghoul",
          "asset": "Jewelevil Bronze Of Goddesses",
          "name": "Enid Rose",
        },
        Player {
          "alias": "Skulldart",
          "asset": "Yellow Orichalcum Of Paladins",
          "name": "Esmeralda Carrillo",
        },
      ],
      "savePlayerSelection": [MockFunction],
    }
  }
>
  <Spring
    config={
      Object {
        "friction": 26,
        "tension": 170,
      }
    }
    force={false}
    from={
      Object {
        "maxHeight": "0",
        "opacity": 0,
      }
    }
    immediate={false}
    inject={[Function]}
    native={false}
    reset={false}
    to={
      Object {
        "maxHeight": "auto",
        "opacity": 1,
      }
    }
  >
    <styled.section
      style={
        Object {
          "maxHeight": "0",
          "opacity": 0,
        }
      }
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "lastClassName": "cPleJF",
              "rules": Array [
                "
    border: 1px solid var(--secondary-color);
    grid-area: details;
    padding: 1em;
    overflow: hidden;
",
              ],
            },
            "displayName": "styled.section",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "section",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        style={
          Object {
            "maxHeight": "0",
            "opacity": 0,
          }
        }
      >
        <section
          className="sc-bwzfXH cPleJF"
          style={
            Object {
              "maxHeight": "0",
              "opacity": 0,
            }
          }
        >
          <Component>
            <styled.h1>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": true,
                      "lastClassName": "hdRpnz",
                      "rules": Array [
                        "
    font-size: 2em;
    font-weight: normal;
    margin: 0;
",
                      ],
                    },
                    "displayName": "styled.h1",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "h1",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <h1
                  className="sc-bdVaJa hdRpnz"
                >
                  Details
                </h1>
              </StyledComponent>
            </styled.h1>
          </Component>
          <styled.ul>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": true,
                    "lastClassName": "bWYfNF",
                    "rules": Array [
                      "
    font-size: 1.5em;
    list-style: none;
    margin: 1em 0;
    padding: 0;
",
                    ],
                  },
                  "displayName": "styled.ul",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "ul",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <ul
                className="sc-htpNat bWYfNF"
              >
                <styled.li>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": true,
                          "lastClassName": "MPdNl",
                          "rules": Array [
                            "
    &:before {
        content: '—';
        padding-right: 0.5em
    }
",
                          ],
                        },
                        "displayName": "styled.li",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <li
                      className="sc-bxivhb MPdNl"
                    >
                      Brianna Forbes
                    </li>
                  </StyledComponent>
                </styled.li>
                <styled.li>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": true,
                          "lastClassName": "MPdNl",
                          "rules": Array [
                            "
    &:before {
        content: '—';
        padding-right: 0.5em
    }
",
                          ],
                        },
                        "displayName": "styled.li",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <li
                      className="sc-bxivhb MPdNl"
                    >
                      Dreamlurk The Unstoppable
                    </li>
                  </StyledComponent>
                </styled.li>
                <styled.li>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bxivhb",
                          "isStatic": true,
                          "lastClassName": "MPdNl",
                          "rules": Array [
                            "
    &:before {
        content: '—';
        padding-right: 0.5em
    }
",
                          ],
                        },
                        "displayName": "styled.li",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bxivhb",
                        "target": "li",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <li
                      className="sc-bxivhb MPdNl"
                    >
                      Foghammer Lead
                    </li>
                  </StyledComponent>
                </styled.li>
              </ul>
            </StyledComponent>
          </styled.ul>
        </section>
      </StyledComponent>
    </styled.section>
  </Spring>
</Component>
`;
